File: chat/src/main/kotlin/chatox/chat/api/response/UserResponse.kt
package chatox.chat.api.response

import chatox.chat.model.ImageUploadMetadata
import java.time.ZonedDateTime

data class UserResponse(
        val id: String,
        val firstName: String,
        val lastName: String?,
        val slug: String?,
        val avatarUri: String?,
        val deleted: Boolean,
        val bio: String?,
        val lastSeen: ZonedDateTime?,
        val dateOfBirth: ZonedDateTime?,
        val createdAt: ZonedDateTime?,
        val online: Boolean?,
        val avatar: UploadResponse<ImageUploadMetadata>?,
        val anonymous: Boolean
)

File: reports/src/users/types/responses/user.response.ts
import {UploadResponse} from "../../../uploads/types";

export interface UserResponse {
    id: string;
    firstName: string;
    lastName?: string;
    bio?: string;
    dateOfBirth?: string;
    slug?: string;
    avatarUri?: string;
    deleted: boolean;
    createdAt: string;
    online: boolean;
    lastSeen?: string;
    avatar?: UploadResponse;
}
